package Epsilon5;

message Player {
    required int32 id = 1;

    required double x = 2;
    required double y = 3;

    required double vx = 4;
    required double vy = 5;

    required double angle = 6;

    optional string name = 7;
}

message World {
    repeated Player players = 1;
}

message Control {

    message KeyStatus {
        required bool keyUp = 1;
        required bool keyDown = 2;
        required bool keyLeft = 3;
        required bool keyRight = 4;
    }

    message MouseClick {
        required double x = 1;
        required double y = 2;
        optional bool buttonRight = 3;
    }

    required KeyStatus keyStatus = 1;
    required double angle = 2;
    optional MouseClick mouseClick = 3;
}
